docs/extensibility/custom-resources.md

```diff
@@ -0,0 +1,432 @@
+
+```dotnetcli
+# Creates a ASP.NET Core project that uses minimal APIs.
+dotnet new webapi --use-minimal-apis --no-openapi -o MailDevResource.NewsletterService
+
+# Add the newsletter service project as a reference to the app host.
+dotnet add .\MailDevResource.AppHost\MailDevResource.AppHost.csproj reference .\MailDevResource.NewsletterService\MailDevResource.NewsletterService.csproj
+
+# Add the newsletter service project to the solution
+dotnet sln .\MailDevResource.sln add .\MailDevResource.NewsletterService\MailDevResource.NewsletterService.csproj
+```
+
+Inside the `Program.cs` in the `MailDevResource.AppHost` project, update the source file to look like the following:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+var maildev = builder.AddMailDev("maildev");
+builder.AddProject<Projects.MailDevResource_NewsletterService>("newsletterservice")
+       .WithReference(maildev);
+
+builder.Build().Run();
+```
+
+After updating the `Program.cs` source file launch the AppHost again and verify that the .NET project that was added started and that the environment variable `ConnectionStrings__maildev` was added to the process (click on the _Details_ link on the `newsletterservice` row in the dashboard).
+
+![Environment variables for Newsletter Service in .NET Aspire Dashboard](./media/maildev-envvar.png)
+
+### Step 4b: Use connection string to send message.
+
+To use the SMTP connection details that were injected into the newsletter service project we will inject an instance of <xref:System.Net.Mail.SmtpClient> into the dependency injection container as a singleton. Add the following code to the `Program.cs` file in the `MailDevResource.NewsletterService` project to setup the singleton service.
+
+```csharp
+builder.Services.AddSingleton<SmtpClient>((sp) =>
+{
+    var smtpUri = new Uri(builder.Configuration.GetConnectionString("maildev")!);
+    var smtpClient = new SmtpClient(smtpUri.Host, smtpUri.Port);
+    return smtpClient;
+});
+```
+
+To test the client we will add two simple subscribe and unsubscribe GET methods to the newsletter service. Add the following code to the `Program.cs` file in the `MailDevResource.NewsletterService` project to setup the ASP.NET Core routes:
+
+```csharp
+app.MapGet("/subscribe", async (SmtpClient smtpClient, string email) => {
+    var message = new MailMessage("newsletter@yourcompany.com", email);
+    message.Subject = "Welcome to our newsletter!";
+    message.Body = "Thank you for subscribing to our newsletter!";
+    await smtpClient.SendMailAsync(message);
+});
+
+app.MapGet("/unsubscribe", async (SmtpClient smtpClient, string email) => {
+    var message = new MailMessage("newsletter@yourcompany.com", email);
+    message.Subject = "You are unsubscribed from our newsletter!";
+    message.Body = "Sorry to see you go. We hope you will come back soon!";
+    await smtpClient.SendMailAsync(message);
+});
```

Comment:
If I recall correctly, the `MailMessage` type is disposable, no? Let's slap a `using` on that, and fix the opening curly brace.

```suggestion
app.MapGet("/subscribe", async (SmtpClient smtpClient, string email) =>
{
    using var message = new MailMessage("newsletter@yourcompany.com", email);
    message.Subject = "Welcome to our newsletter!";
    message.Body = "Thank you for subscribing to our newsletter!";
    await smtpClient.SendMailAsync(message);
});

app.MapGet("/unsubscribe", async (SmtpClient smtpClient, string email) =>
{
    using var message = new MailMessage("newsletter@yourcompany.com", email);
    message.Subject = "You are unsubscribed from our newsletter!";
    message.Body = "Sorry to see you go. We hope you will come back soon!";
    
    await smtpClient.SendMailAsync(message);
});
```